subprojects/wrapper-shared/src/main/java/org/gradle/wrapper/Download.java

```diff
@@ -84,6 +85,25 @@ private void configureProxyAuthentication() {
         }
     }
 
+    public void sendHeadRequest(URI uri) throws Exception {
+        URL safeUrl = safeUri(uri).toURL();
+        int responseCode = -1;
+        try {
+            HttpURLConnection conn = (HttpURLConnection)safeUrl.openConnection();
```

Comment:
:x: I think the test connection should be done directly on the provided URI instead of the "safe URI" that is stripped from credentials. Otherwise this will always fail for folks using authenticated URLs here. The "safe URI" should still be used for error reporting though.

Reply:
The "safe URI" is what's currently used for downloading (line 117), shouldn't this be consistent? I agree that it should be used for error reporting.

Reply:
Let's keep it as it is then.